pkg/okteto/preview.go

```diff
@@ -121,83 +139,74 @@ type previewIDStruct struct {
-			})
-		}
-		queryVariables := map[string]interface{}{
-			"name":       graphql.String(name),
-			"scope":      PreviewScope(scope),
-			"repository": graphql.String(repository),
-			"branch":     graphql.String(branch),
-			"sourceURL":  graphql.String(sourceUrl),
-			"variables":  variablesVariable,
-			"filename":   graphql.String(filename),
-		}
-		err := mutate(ctx, &mutationStruct, queryVariables, c.client)
-		if err != nil {
-			return nil, c.translateErr(err, name)
-		}
-		previewResponse.Action = &types.Action{
-			ID:     string(mutationStruct.Response.Action.Id),
-			Name:   string(mutationStruct.Response.Action.Name),
-			Status: string(mutationStruct.Response.Action.Status),
-		}
-		previewResponse.Preview = &types.Preview{
-			ID: string(mutationStruct.Response.Id),
-		}
-	} else {
-		mutationStruct := deployPreviewMutation{}
-		queryVariables := map[string]interface{}{
-			"name":       graphql.String(name),
-			"scope":      PreviewScope(scope),
-			"repository": graphql.String(repository),
-			"branch":     graphql.String(branch),
-			"sourceURL":  graphql.String(sourceUrl),
-			"filename":   graphql.String(filename),
-			"variables": []InputVariable{
-				{Name: graphql.String("OKTETO_ORIGIN"), Value: graphql.String(origin)},
-			},
-		}
-		err := mutate(ctx, &mutationStruct, queryVariables, c.client)
-		if err != nil {
-			return nil, c.translateErr(err, name)
-		}
-		previewResponse.Action = &types.Action{
-			ID:     string(mutationStruct.Response.Action.Id),
-			Name:   string(mutationStruct.Response.Action.Name),
-			Status: string(mutationStruct.Response.Action.Status),
+func (c *previewClient) getDeployVariables(name, scope, repository, branch, sourceUrl, filename string, variables []types.Variable, labels []string) map[string]interface{} {
+	variablesVariable := make([]InputVariable, 0)
+	for _, v := range variables {
+		variablesVariable = append(variablesVariable, InputVariable{
+			Name:  graphql.String(v.Name),
+			Value: graphql.String(v.Value),
+		})
+	}
+	injectOrigin := true
+	for _, v := range variablesVariable {
+		if v.Name == "OKTETO_ORIGIN" {
+			injectOrigin = false
 		}
-		previewResponse.Preview = &types.Preview{
-			ID: string(mutationStruct.Response.Id),
+	}
+	if injectOrigin {
+		origin := config.GetDeployOrigin()
+		variablesVariable = append(variablesVariable, InputVariable{
+			Name:  graphql.String("OKTETO_ORIGIN"),
+			Value: graphql.String(origin),
+		})
+	}
+	vars := map[string]interface{}{
+		"name":       graphql.String(name),
+		"scope":      PreviewScope(scope),
+		"repository": graphql.String(repository),
+		"branch":     graphql.String(branch),
+		"sourceURL":  graphql.String(sourceUrl),
+		"variables":  variablesVariable,
+		"filename":   graphql.String(filename),
+	}
+
+	if len(labels) > 0 {
+		labelsVariable := make(labelList, 0)
+		for _, l := range labels {
+			labelsVariable = append(labelsVariable, graphql.String(l))
 		}
+		vars["labels"] = labelsVariable
```

Comment:
do I understand correctly that this request now will send `labels` but we also need a change on the backend to receive them and store them in k8s?